### DISCUSSION TITLE
Video playlists: Does it make sense to pre-convert?

### DISCUSSION THREAD

**Discussion by Moonbase59** [savonet/liquidsoap#3147] [Q&A] [answered]
I’m using a video playlist for a webradio station. It has only a few (~10) videos which are in different formats, i.e. 1080p60, 1080p30, some even with 24000/1001 fps.

I’m outputting as ffmpeg 720p25 MPEG-TS, with AC3 sound.

In order to minimize CPU load on the radio station VM, does it make sense to _pre-convert_ the source material into 720p25 format, or is Liquidsoap/ffmpeg fast enough that it doesn’t really matter?

If pre-converting, what format and container type would you suggest for optimal throughput with lowest CPU usage?

My `output` is currently:
```
output.icecast(%ffmpeg(format="mpegts",
  %audio(codec="ac3",channel_coupling=0),
  %video(codec="libx264",b="2600k",
         "x264-params"="scenecut=0:open_gop=0:min-keyint=150:keyint=150",
         preset="ultrafast")),
host="radio", port=8000, password="mysecretpassword", mount="/video.m2t", send_icy_metadata=true, public=false, s)
```

If all goes well, I plan to eventually add a HLS or DASH output to accomodate for various endpoint bandwidth requirements.

**Comment by Moonbase59** [comment, marked as answer]
Answering my own question, after watching closely what happens:

- _No conversion_ seems necessary if the source material is max. 1080p, at a **max. framerate of 30 fps**.
- _Pre-converting_ seems absolutely mandatory on **60 fps material**, because LS simply can’t keep up when only minimal extra processing is required, like adding a logo, text or the like. LS will almost immediately start with "need to catch up" errors.

**Comment by gAlleb** [comment]
@Moonbase59 Hi! Figured out that maybe I could address to you here since no other means of communication are available)

I like how you're implementing your videostream. And I've noticed an interesting feature that you've added: the name of current playlist which is shown above the title. And I'd like to 'steal' it :) since i don't understand how to implement it myself. Would you be so kind to share the `function/code of adding dynamic playlist name` if it's ok? (Want to try it out with Azuracast)

Thanks!

**Reply by Moonbase59** [reply]
The code is still a mess (though it works stable for me since months). I plan to make it it available, either on GitHub or at least in a Gist, but let me clean some things up a little first, please. So many things to do, never enough time… Know how it is…

**Reply by gAlleb** [reply]
@Moonbase59 Sure thing. 
Thank you! Looking forward!

**Reply by Moonbase59** [reply]
Just tag me if it tages _too_ long, I’m just writing [azuracast_xmltv](https://github.com/Moonbase59/azuracast_xmltv), to generate EPG data from AzuraCast stations ;-)
